Add steel buttons recipe, update other button recipes.

Tamiore opened the pull request:
#### Summary
Content "Add steel buttons recipe, update other button recipes."

#### Purpose of change

There is no currently no recipe for `steel button`. Recipes for `wooden button` and  `plastic button` are unreasonably restrictive in some areas.

#### Describe the solution

Added a recipe for `steel buttons`.
Allow for `PUNCH 1` as an alternative to `DRILL 1` in  `wooden button` and  `plastic button`. (Plastic is softened by heat, a hole in a thin piece of wood can be made with a knife and punch just fine.) 
Remove `FILE 1` from `wooden button`. ( `FILE 1` is barely implemented. )
Add `CUT 1` to  `wooden button`. (It's a carving recipe after all!)
Allow for alternative lubricants in `wooden button` recipe (I assume that's what `wax` is in this recipe).

#### Testing

.json change, works fine locally.

#### Additional context

Fun fact: there is currently no way for a blacksmith to make a tool with `FILE`. Should probably be looked at, at  `FILE 1/2` has already appeared in a few recipes.

actual-nh commented:
Ping: @Atsumari?

Atsumari commented:
I'll take a look at making the file system more diverse.

Maybe add them to sword recipes and/or small items.

actual-nh commented:
> I'll take a look at making the file system more diverse.

Thanks!

> Maybe add them to sword recipes and/or small items.

How does the recipe for steel buttons look, BTW?

Tamiore commented:
> I'll take a look at making the file system more diverse.

Right now the big issue is that you can't craft anything with a `FILE` rating. Neither at blacksmithing tech level, nor before it (glue same sand to board or something?).

actual-nh commented:
> Right now the big issue is that you can't craft anything with a `FILE` rating. Neither at blacksmithing tech level, nor before it (glue some `sand` to surface or something?).

That's more complicated than it looks; the glue gets rather tricky, particularly if you want to glue the sand to paper or something else flexible.

Tamiore commented:
If we assume `duct tape` crafting recipe is reasonable, probably the same recipe with added `sand` if reasonable for a sandpaper?

anothersimulacrum commented:
Please don't remove filing where it makes sense. There are improvements that can be made to it's implementation, but let's make them, not ignore them.

Tamiore commented:
> Please don't remove filing where it makes sense. There are improvements that can be made to it's implementation, but let's make them, not ignore them.

I have no problem with adding filing back to wooden buttons. But _does_ it make sense on them, seeing that they are carved from pretty soft material (wood) and also lubricated?

Maleclypse reviewed (changes requested):
The wax is used to treat the wood.  None of the items you added in lines 662-667 will treat the wood.  There are alternatives to wax for treating wood but they aren't currently implemented in game.

Maleclypse commented:
> I have no problem with adding filing back to wooden buttons. But _does_ it make sense on them, seeing that they are carved from pretty soft material (wood) and lubricated on top of that?

Not lubricated. Finished to prevent splintering.

Tamiore commented:
>The wax is used to treat the wood. None of the items you added in lines 662-667 will treat the wood. There are alternatives to wax for treating wood but they aren't currently implemented in game.

Elaborate, please.
Here I can see a varnish made of rosin and linseed oil: https://simplifier.neocities.org/varnish.html
And, from what I understand, any type of drying oil should work here, not necessarily linseed oil specifically. 
So would some kind of drying oil alone be enough to "treat" those buttons?
Would drying oil + rosin be enough?

Maleclypse commented:
> > The wax is used to treat the wood. None of the items you added in lines 662-667 will treat the wood. There are alternatives to wax for treating wood but they aren't currently implemented in game.
> 
> Elaborate, please. @Maleclypse
> Here I can see a varnish made of rosin and linseed oil: https://simplifier.neocities.org/varnish.html
> And, from what I understand, any type of drying oil should work here, not necessarily linseed oil specifically.
> 
> So, given C:DDA's level of abstraction, would rosin alone be enough to "treat" those buttons (seeing how wax alone is)?
> Would some kind of drying oil alone be enough?
> Would drying oil + rosin be enough?

Yes. Rosin should be enough, I'd forgotten it was in game.  Lard and other animal fats would just make the wood sticky.  Does that answer your question?

Tamiore commented:
>Yes. Rosin should be enough, I'd forgotten it was in game. Lard and other animal fats would just make the wood sticky. Does that answer your question?

It does.
Added rosin, removed everything else other than wax.

Maleclypse reviewed (commented):
Thanks for making the rosin change to the wooden buttons.   I don't know anything about making steel buttons.

Atsumari commented:
I'm assuming we are talking about clothes buttons? If that's the case you'd just need a small metal punch or drill and some metal files.

Tamiore commented:
>I'm assuming we are talking about clothes buttons? If that's the case you'd just need a small metal punch or drill and some metal files.

The current recipe was envisioned for hot-working the metal, so there are a few other tools involved. 
Does the current (hot-working) recipe included in this PR look sufficient? Or should I add something else, or remove something?

Atsumari commented:
Looks fine.

Maleclypse reviewed (changes requested):
I just noticed the file 1 removal.   File 1 is sandpaper and is necessary to make the buttons ready for finishing.  Cut is not an appropriate replacement for file in this instance.   Also you'd probably need File 1 for smoothing the blacksmithed buttons as well.   There are ancient ways of creating equivalents to sandpaper.  If File 1 scarcity is a worry, I'd google what was used and then add that product.

Atsumari commented:
To be fair you'd only need sandpaper on most items if you want a "finished" look.

Buttons would still be functional it just wouldn't look polished.

Tamiore commented:
>If File 1 scarcity is a worry, I'd google what was used and then add that product.

That's exactly the problem. Because right now you can't _craft_ anything with `FILE 1`

So I google a bit and found this: https://craftknights.com/10-ways-to-smooth-wood-without-sandpaper/

Look at /#1 there, I added a new tool: sandleather. (Literally just some sand with leather.)

actual-nh commented:
> To be fair you'd only need sandpaper on most items if you want a "finished" look.
> 
> Buttons would still be functional it just wouldn't look polished.

Functional except for possibly getting a splinter or cutting oneself (the latter if using a metal punch, I would guess)?

Tamiore commented:
Anything else seems off about the three buttons recipes (wooden/plastic/steel)?
Might as well update it while I'm still at it.

Maleclypse commented:
I don’t see any other changes needed but if there are hopefully someone will see it before it’s merged. If you have any questions about merging process feel free to ask. :). Thanks for doing this!

I-am-Erk commented on data/json/recipes/tools/tools_primitive.json:
```diff
@@ -335,5 +335,18 @@
+    "time": "5 s",
+    "reversible": true,
+    "autolearn": true,
+    "components": [ [ [ "material_sand", 1 ] ], [ [ "leather", 1 ] ] ]
```
I can't find any references to this stuff IRL, how is this made? What makes the sand stick to the leather to be used for filing?

Maleclypse replied:
Now that you mention it, I think you put a container under whatever your sanding and then put sand on top of it then pull the leather back on forth on it and reuse the sand from the container below.

Tamiore replied:
This was my source: https://craftknights.com/10-ways-to-smooth-wood-without-sandpaper/
Nothing really holds the sand as far as I understand the source. You just apply it to the surface being polished as needed and rub with the leather.

I-am-Erk replied:
I see. I guess this is an okay solution.
